script.sql

```diff
@@ -1,147 +1,85 @@
 );
-CREATE TABLE App  ( 
-id INT,
-name varchar(60),
-publisherId INT,
-categoryId INT,
-/*FOREIGN KEY (publisherId) REFERENCES Publisher(id) ON DELETE CASCADE ON UPDATE CASCADE,
-FOREIGN KEY (categoryId) REFERENCES category(id) ON DELETE CASCADE ON UPDATE CASCADE, */
-description VARCHAR(255)
 
+CREATE TABLE App  ( 
+        id INT,
+        name VARCHAR(60),
+        publisherId INT,
+        categoryId INT,
+        description VARCHAR(255),
+        PRIMARY KEY (id),
+        FOREIGN KEY (publisherId) REFERENCES Publisher(id) 
+                ON DELETE SET NULL ON UPDATE NO ACTION,
+        FOREIGN KEY (categoryId) REFERENCES Category(id) 
+                ON DELETE SET NULL ON UPDATE NO ACTION
 );
-CREATE TABLE  AppVersion  ( 
-appId INT,
-version char(10),
-
-releaseDate DATETIME,
-rating INT,
-price Decimal(10,2), 
-description VARCHAR(500)
 
+CREATE TABLE  AppVersion  ( 
+        appId INT,
+        version CHAR(10),
+        releaseDate DATETIME,
+        rating INT,
+        price DECIMAL(10,2), 
+        description VARCHAR(500),
+        PRIMARY KEY (appId, version),
+        FOREIGN KEY (appId) REFERENCES App(id)
+                ON DELETE CASCADE ON UPDATE CASCADE
 );
+
 CREATE TABLE AppVersionReview  ( 
-appId INT,
-PRIMARY KEY  (appId),
-version CHAR(10),
-reviewer CHAR(20),
-reviewDate DATETIME,
-rating INT,
-review VARCHAR(1000));
+        appId INT,
+        version CHAR(10),
+        reviewer CHAR(20),
+        reviewDate DATETIME NOT NULL,
+        rating INT,
+        review VARCHAR(1000),
+        PRIMARY KEY (appId, version, reviewer),
+        FOREIGN KEY (appId, version) REFERENCES AppVersion(appId, version)
+                ON DELETE CASCADE ON UPDATE CASCADE
+);
+
+-- INSERT
+INSERT INTO Publisher VALUES 
+(1,'Adobe'),
+(2,'Ubisoft');
 
-/*Insert*/
-INSERT INTO Publisher (id,name)
- VALUES (1,'Adobe'),
-        (2,'Ubisoft');
-INSERT INTO Category(id,name,parentId)
-VALUES  (1,'Photo and Video',null),
+INSERT INTO Category VALUES  
+(1,'Photo and Video',null),
 (2,'Editing photos and videos',1),
 (3,'Games',null),
 (4,'Strategy Games',3);
 
-INSERT INTO App (id,name,publisherId,categoryId,description)
- VALUES (100, 'Photoshop', 1, 2, 'Bring your pictures to life'),
+INSERT INTO App VALUES 
+(100, 'Photoshop', 1, 2, 'Bring your pictures to life'),
 (200, 'Hungry Shark World', 2, 3, 'Endless hunt, eat to survive!');
-INSERT INTO AppVersion (appId,version,releaseDate,rating,price,description)
- VALUES (100, '1.0', '2008-09-11', 1, 4.99, 'First version'),
+
+INSERT INTO AppVersion VALUES 
+(100, '1.0', '2008-09-11', 1, 4.99, 'First version'),
 (100, '2.0', '2020-01-01', 5, 19.99, 'Premium version for more money'),
 (200, '5.0', '2018-05-23', 3, 1.99, null),
 (200, '6.1', '2020-07-11', 5, 0.99, 'Have fun!');
 
 
-
-        /* DELETE*/
+-- DELETE
 DELETE FROM  AppVersionReview WHERE rating <= 3;
-DELETE FROM  Publisher WHERE id <= 3;
-
+DELETE FROM  App WHERE publisherId IN (SELECT id FROM Publisher WHERE name = 'Adobe');
```

Comment:
Your statement will delete everything. Here is a more generalized and right approach.